### ISSUE TITLE
Regarding taxonomy assignment

### ISSUE THREAD

**Issue by ShailNair** [moyn413/nifHdada2#2]
Hi, 
Recently we amplified  nifH gene from a mixed bacterial sample and sequenced the resultant band. I used DADA2 workflow with default parameters and for taxonomic annotation used your database (as the default 16S SILVA database failed to assign taxonomy beyond domain). Using your database the workflow was able to classify many sequences till genus level but I see many of the assigned taxa are unexpected for us. Before we had done 16s amplicon sequencing, culture-dependent isolation and identification, and metagenomic analysis (using NCBI NR database for classification) on the same sample. A large number of taxa (assigned using nifHdada2 database) are different from what we have found using the other methods. 

Almost 80-90% of the sequences are classified as Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis , which in other methods was not present. Many genera Belonging to rhizobiales (which we expected in our sample infered from other analysis) did not show up here.
ASV | Kingdom | Phylum | Class | Order | Family | Genus
-- | -- | -- | -- | -- | -- | --
ASV_1 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_2 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_3 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_4 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_5 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_6 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_7 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_8 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_9 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_10 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_11 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_12 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_13 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_14 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_15 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_16 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_17 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_18 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_19 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_20 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_21 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_22 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_23 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_24 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_25 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_26 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_27 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_28 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_29 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_30 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_31 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_32 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_33 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis
ASV_34 | Bacteria | Proteobacteria | Betaproteobacteria | Nitrosomonadales | Sterolibacteriaceae | Methyloversatilis

I didn't mean to discourage/falsify your work. Just want to know your take on this, as you have worked on nifH gene before. Is there any other nifH related database (that you came across while analyzing your samples)which I can use for comparison.

**Comment by moyn413** [comment]
Hi @razz1618, thank you so much for your feedback. This is a new database, so I'm happy to know about issues and try to help improve it! It makes sense that the 16S SILVA database was not able to assign taxonomy to the nifH gene, as these are different regions. Based on my own data, I find that things that show up in the nifH amplicon data, also show up in my 16S amplicon data from the same samples, using the nifHdada2 and 16S SILVA databases, respectively, for taxonomic assignment.

Could you provide me with a bit more information? 

-What version of the nifHdada2 database did you use? 
-What were the bootstrap values associated with the taxonomic assignments of Methyloversatilis? 
-Are the genera of Rhizobiales you identified using other methods also present in the database? 

Regarding other databases available, there is one that was shared in a github thread. (https://github.com/benjjneb/dada2/issues/631). The file name is "seqDatabase1.fasta.txt".

**Comment by ShailNair** [comment]
Hi, 
1.I am using version 1.1.0
2. I didn't change the default bootstrap (minBoot = 50) for the taxonomy assignment as I was happy with the same bootstrap for my 16s amplicon analysis. 
3. yes. some of them are there. but not detected in my sample. in fact, some other genus which I expected didn't show up. It might be due to sequencing or maybe the post-analysis.

And my bad, genus related to Nitrosomonadales are present in my metagenome, but at lower abundance.

**Comment by moyn413** [comment]
The nifH database is much smaller than the 16S SILVA database, and as a result, the taxonomic assignment will be on average, less good and with lower bootstrap values. I'm sure you'll find that the assign taxonomy step is very quick in comparison with SILVA! Out of curiosity, you could remove the bootstrap threshold when you are assigning taxonomy and then filter out low bootstrap values later on in your processing. 

I'm not sure what type of 16S metagenomic sample you are working with, but if it is environmental, it wouldn't be too surprising to have your nitrogen fixers be in lower abundance.  I also believe that not all Rhizobiales have nifH (https://www.biorxiv.org/content/10.1101/789107v1.full)

**Comment by ShailNair** [comment]
Yes, of course not all  Rhizobiales will have nif. My query was regarding those genera/species which I have confirmed. We did 16s, metagenomics and culture-dependent isolation and amplification of nifH gene. Bootstrap values might be worth checking (I will check let you know). 

the sample is not exactly environmental but the experimental setup should mimic natural habitat (bacterial community co-cultured with algae in natural sterile sea water). But still there will be some taxa dominating the system as the setup is in a microcosm state.
